mtraining/src/main/java/org/motechproject/mtraining/service/impl/MTrainingServiceImpl.java

```diff
@@ -174,8 +172,21 @@ public Lesson createLesson(Lesson lesson) {
 
     @Override
     @Transactional
-    public List<Lesson> getLessonByName(String lessonName) {
-        return lessonDataService.findLessonByName(lessonName);
+    public List<Lesson> getLessonsByName(String lessonName) {
+        return lessonDataService.findLessonsByName(lessonName);
+    }
+
+    @Override
+    public List<Lesson> getLessonsByProperties(Map<String, String> properties) {
+        List lessons = lessonDataService.executeQuery(new QueryExecution<List<Lesson>>() {
+            @Override
+            public List<Lesson> execute(Query query, InstanceSecurityRestriction restriction) {
```

Comment:
We should create a separate, generic class for this execution, since the body of the method is always the same.